app/modules/users/models.py

```diff
@@ -384,10 +373,8 @@ def get_account_recovery_code(self):
 
     def set_password(self, password=None):
         if password is None:
-            # Generate a random password for the user
-            from app.modules.auth.models import _generate_salt
-
-            password = _generate_salt(128)
+            # This function "sets" the password, it's the responsibility of the caller to ensure it's valid
```

Comment:
If password is not optional anymore, lets drop the default value for password in `password=None` in the argument of the function signature